benchmarks/bench_plot_randomized_svd.py

```diff
@@ -323,8 +323,9 @@ def norm_diff(A, norm=2, msg=True, random_state=None):
 
 
 def scalable_frobenius_norm_discrepancy(X, U, s, V):
-    # if the input is not too big, just call scipy
-    if X.shape[0] * X.shape[1] < MAX_MEMORY:
+    if not sp.sparse.issparse(X) or (X.size * X.dtype.itemsize < MAX_MEMORY):
```

Comment:
Given your other comment, I think this needs to be `X.shape[0] * X.shape[1]`. For a sparse matrix `X.size` is the number of stored values.

(Also with `X.size`, I run into a seg fault with `big sparse matrix 1000000 x 10000`)

Reply:
OK thanks, I reverted to `X.shape[0] * X.shape[1]`